File: app/src/main/java/com/juawapps/newsspread/di/base/AppModuleDatabase.java
package com.juawapps.newsspread.di.base;

import android.content.Context;

import com.juawapps.newsspread.data.db.AppDatabase;
import com.juawapps.newsspread.data.db.ArticleDao;
import com.juawapps.newsspread.data.db.DatabaseProvider;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Database module
 */
@Module
public class AppModuleDatabase {

    @Provides
    @Singleton
    AppDatabase provideNewsDatabase(Context applicationContext) {
        return DatabaseProvider.getDatabase(applicationContext);
    }

    @Singleton
    @Provides
    ArticleDao provideUserDao(AppDatabase database) {
        return database.articleDao();
    }

}

File: app/src/main/java/com/juawapps/newsspread/di/base/AppModuleDatabase.kt
package com.juawapps.newsspread.di.base

import android.content.Context

import com.juawapps.newsspread.data.db.AppDatabase
import com.juawapps.newsspread.data.db.ArticleDao
import com.juawapps.newsspread.data.db.DatabaseProvider

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Database module
 */
@Module
class AppModuleDatabase {

    @Provides
    @Singleton
    internal fun provideNewsDatabase(applicationContext: Context): AppDatabase {
        return DatabaseProvider.getDatabase(applicationContext)
    }

    @Singleton
    @Provides
    internal fun provideUserDao(database: AppDatabase): ArticleDao {
        return database.articleDao()
    }

}
